src/psyclone/tests/gocean1p0_transformations_test.py

```diff
@@ -1184,3 +1184,98 @@ def test_module_inline_warning_no_change():
     inline_trans = KernelModuleInlineTrans()
     _, _ = inline_trans.apply(kern_call, inline=False)
+
+
+def test_loop_swap_correct():
+    ''' Testing correct loop swapping transform. Esp. try first, middle, and
+    last invokes to make sure the inserting of the inner loop happens at
+    the right place.'''
+
+    psy, invoke = get_invoke("test27_loop_swap.f90", 0)
+    invoke = psy.invokes.get("invoke_loop1")
+    schedule = invoke.schedule
+    schedule_str = str(schedule)
+
+    # First make sure to throw an early error if the source file
+    # test27_loop_swap.f90 should have been changed
+    expected_schedule = '''Loop[]: j= lower=2,jstop,1
+Loop[]: i= lower=2,istop,1
+kern call: bc_ssh_code'''
+    assert expected_schedule in  schedule_str
+
+    expected_schedule = '''Loop[]: j= lower=1,jstop+1,1
+Loop[]: i= lower=1,istop,1
+kern call: bc_solid_u_code'''
+    #assert expected_schedule in  schedule_str
+    print expected_schedule, "\n++++++++\n",schedule_str
+
+    expected_schedule = '''Loop[]: j= lower=1,jstop,1
+Loop[]: i= lower=1,istop+1,1
+kern call: bc_solid_v_code'''
+    assert expected_schedule in schedule_str
+
+    # Now swap the first loops
+    swap = LoopSwapTrans()
+    swapped1, _ = swap.apply(schedule.children[0])
+    psy.invokes.get('invoke_loop1').schedule = swapped1
+    schedule_str = str(swapped1)
+
+    expected_schedule = '''Loop[]: i= lower=2,istop,1
+Loop[]: j= lower=2,jstop,1
+kern call: bc_ssh_code'''
+    assert expected_schedule in schedule_str
+
+    # Now swap the middle loops
+    swapped2, _ = swap.apply(swapped1.children[1])
+    psy.invokes.get('invoke_loop1').schedule = swapped2
+    schedule_str = str(swapped2)
+
+    expected_schedule = '''Loop[]: i= lower=1,istop,1
+Loop[]: j= lower=1,jstop+1,1
+kern call: bc_solid_u_code'''
+    assert expected_schedule in schedule_str
+
+    # Now swap the middle loops
+    swapped3, _ = swap.apply(swapped2.children[2])
+    psy.invokes.get('invoke_loop1').schedule = swapped3
+    schedule_str = str(swapped3)
+
+    expected_schedule = '''Loop[]: i= lower=1,istop+1,1
+Loop[]: j= lower=1,jstop,1
+kern call: bc_solid_v_code'''
+    assert expected_schedule in schedule_str
+    return
+
+
+def test_loop_swap_errors():
+    ''' Test loop swapping transform with incorrect parameters. '''
+
+    psy, invoke = get_invoke("test27_loop_swap.f90", 1)
+
+    schedule = invoke.schedule
+    swap = LoopSwapTrans()
+
+    # Test error if given node is not the outer loop of an at least
+    # double nested loop:
+    with pytest.raises(TransformationError) as error:
+        swap.apply(schedule.children[0].children[0])
+    assert "First inner statement is not a loop" in str(error.value)
+
+    # Not a loop:
+    with pytest.raises(TransformationError) as error:
+        swap.apply(schedule.children[0].children[0].children[0])
+    assert "Given node is not a loop" in str(error.value)
+
+    # Now create an outer loop with more than one inner statements
```

Comment:
Should this say "...with more than one inner loop?" i.e. they're not perfectly nested?